informational overlay:Redesign UI of hotkeys info modal to modern one

Taiki-Choda opened the pull request:
Improve UI of the keyboard shortcuts info modal.
Mainly, reducing using borders.

## Before
<img src="https://user-images.githubusercontent.com/86971544/201510948-0a059b02-d0f5-4506-bd3e-f1ed172f08f9.jpg" width="45%">

## After
<img src="https://user-images.githubusercontent.com/86971544/201511059-86edf67c-506a-4e84-81d8-6dfffffa65ec.jpg" width="35%"><img src="https://user-images.githubusercontent.com/86971544/201511065-1d9b7a74-a583-43b6-8071-b15cea4f2b40.jpg" width="35%">
<img src="https://user-images.githubusercontent.com/86971544/201511063-797f173d-225a-4f93-8e07-251319419a4e.jpg" width="35%"><img src="https://user-images.githubusercontent.com/86971544/201511060-457d1f38-d6fe-45bb-871d-1d3b8abb880b.jpg" width="35%">

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.
- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

alya commented:
Thanks! I like overall direction, but a few thoughts on how to iterate:

1. The actual keys looks quite small to me within their box. Can we align the design for the keys with #22746?
2. I don't think it makes sense to add the background blurring behind the modal effect just here, as it makes it inconsistent with other overlays (e.g, settings). Let's drop it from the PR; we can have a separate discussion about it #design if you like.
3. The bottom button design doesn't match other buttons in the app. I'd be inclined to just keep it as a link for now, unless @terpimost has a better suggestion.

![Screen Shot 2022-12-12 at 9 57 42 PM](https://user-images.githubusercontent.com/2090066/207238436-0b365e0b-eba6-4a87-b51e-74654f8412f8.png)

Taiki-Choda commented:
@alya 
Here is the fixed commit!
<img src="https://user-images.githubusercontent.com/86971544/207792306-083ffcba-0a35-4689-a27b-5971bd486c75.jpg" width="50%" />

terpimost commented:
I suggest to:
- make color of the hot key in the light theme as #466ae8
- decrease padding inside a hot key (.informational-overlays kbd) to `padding: 1px 0.5em;`
- decrease padding inside a table row to `padding: 0.5rem 0;`
- decrease gap inside the row (.hotkeys__table-row) to `gap: 0 0.4rem;`
- change grit template in a row to `grid-template-columns: 1fr 1.1fr;`
- `.hotkeys__title` to `margin: 0`
- `.hotkeys__table` to  `margin-bottom: 2rem;`
- In search filters make left column narrower 

style of tabs is new... I suggest to stick to the old style unless we want to redesign them, which I guess should be a separate PR?

Taiki-Choda commented:
Thanks for your feedback @terpimost.
I updated this PR. How about this one?

<img src="https://user-images.githubusercontent.com/86971544/209285406-00c0778c-a2f3-423c-a0d4-295e230c3a88.gif" width="50%" />

terpimost commented:
Now we should polish the dark theme 
<img width="776" alt="image" src="https://user-images.githubusercontent.com/1903309/209418941-4f51092d-c470-42bb-86f6-7041d03abec6.png">

Taiki-Choda commented:
Fixed to apply dark theme correctly.
Could you review this?
@terpimost 

<img src="https://user-images.githubusercontent.com/86971544/209454597-f38df2d2-ee5c-45f1-b218-53fa8a9dce92.gif" width="50%" />

alya commented:
Thanks! A few comments from manually testing the PR.

## Keyboard shortcuts tab

1. The "or" between keys looks too faint / hard to read to me in light mode. Can we make it a bit darker?
2. It looks like there's a missing space before "then":
![Screen Shot 2023-01-02 at 2 12 59 PM](https://user-images.githubusercontent.com/2090066/210281358-8538dd5e-ac79-45f1-8da6-cca36bdc5e7f.png)


## Message formatting tab

1. The text in the right column should be the same color as the text on the left, not greyed out.
2. Can we get the keyboard shortcuts left-aligned with the text above? I think that would look better than having them slightly shifted to the right, as the currently are.

alya commented:
Hm, I wonder if the dark theme changes didn't get pushed properly? It still looks broken when I test it:

![Screen Shot 2023-01-02 at 2 18 14 PM](https://user-images.githubusercontent.com/2090066/210281532-3ccf0a3a-a9d2-436c-90e8-a5d0ae888db1.png)

Taiki-Choda commented:
Thanks for your feedback @alya !
I updated the commit so please check this one.

- Text colors were made darker to improve readability.
- Added margin after "then".
- Removed the margin of some keyboard shortcuts in the Message Formatting tab to make them left-aligned.
- The dark theme bug is fixed, at least on my environment.

<img src="https://user-images.githubusercontent.com/86971544/213364416-eed49da7-d311-4179-b033-a38e5971235b.jpg" width="50%" />
<img src="https://user-images.githubusercontent.com/86971544/213364427-5c6f7b48-1178-4136-bd07-f83d1df6a2fe.jpg" width="50%" />

alya commented:
Thanks! There still seems to be something of with the font in the right-side column on the message formatting tab. Is the weight of it really light in some rows?

I think this looks good:
![Screen Shot 2023-01-26 at 2 16 54 PM](https://user-images.githubusercontent.com/2090066/214963514-b09ef2b6-200c-48c9-ac2d-4c67976d1e54.png)

But this looks odd; the font weight should be the same as on the left:

![Screen Shot 2023-01-26 at 2 16 43 PM](https://user-images.githubusercontent.com/2090066/214963552-a112718d-5a91-4fb8-bae0-726273f65b34.png)

alya commented:
Looks good to me otherwise!

Taiki-Choda commented:
@alya 
Now I fixed font-weight issue.

<img src="https://user-images.githubusercontent.com/86971544/225027773-56ea719a-9f36-4925-a37c-c98e33a382a3.jpg" width="40%" />

alya commented:
Cool, thanks! The screenshot looks good, so I think this PR will be ready for a code review from @amanagr once it passes tests.

Taiki-Choda commented:
@amanagr 
Now I have fixed the errors and split the commit for review.

amanagr commented:
I think we can move some rows to be single line like italics and bold, 
<img width="844" alt="Screenshot 2023-03-16 at 1 36 40 PM" src="https://user-images.githubusercontent.com/25124304/225553567-8320d51a-8b75-47e6-99b2-81fb1ccfd27b.png">

Seems simple enough to do too.

alya commented:
Based on the discussion [here](https://chat.zulip.org/#narrow/stream/101-design/topic/keyboard.20shortcuts.20reference/near/1527524), @Taiki-Choda I think this PR might be ready for @timabbott 's review once the merge conflicts have been resolved.

alya commented:
@Taiki-Choda Would you be up for resolving the conflicts?

amanagr commented:
discussion: https://chat.zulip.org/#narrow/stream/101-design/topic/keyboard.20shortcuts.20reference

amanagr commented:
@Ddharmani3 should we close this in favour of  #25979 ?
